### PR TITLE
CHDEXCHAE workpackage

### PR DESCRIPTION
**Pull request by duttonw**

### CODE REVIEW

File: .docker/test.ini [hunk 1]
```diff
@@ -40,7 +40,7 @@ beaker.session.secret = bSmgPpaxg2M+ZRes3u1TXwIcE
 app_instance_uuid = 6e3daf8e-1c6b-443b-911f-c7ab4c5f9605
 
 # repoze.who config
-who.config_file = %(here)s/who.ini
+who.config_file = /srv/app/who.ini
```
Comment (ThrawnCA) [inline .docker/test.ini:43]: I would prefer to leave this as a variable, rather than relying on the Docker container path.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,48 @@
+###############################################################################
```
Comment (ThrawnCA) [inline Makefile:1]: As previously discussed, a Makefile is a very poor approach to dependency installation when `pip` is sufficient for the job, is the standard Python tool on all platforms, and is already configured in this repository.

**Review by ThrawnCA — COMMENTED** [review]
This is cluttered with literally hundreds of "changes" that are just splitting code over two lines to avoid going over a maximum line length, or similar refactoring like replacing function imports with module imports.

It's not necessary, it's arguably not even an improvement (just a different style, with both strengths and weaknesses), it makes actual code review much harder, and it's not what Link Digital is being paid for.

File: ckanext/data_qld/validation.py [hunk 3]
```diff
@@ -21,16 +28,215 @@ def scheming_choices(field, schema):
...
+    alignment_value = resource.extras.get(const.FIELD_ALIGNMENT)
+    resource_schema = resource.extras.get(const.FIELD_RES_SCHEMA)
+
+    if not resource_schema:
+        return False
+
+    if is_url_valid(resource_schema):
+        schemas_aligned = resource_schema == default_schema
+    else:
+        default_schema = json.loads(default_schema) if isinstance(
```
Comment (ThrawnCA) [inline ckanext/data_qld/validation.py:179]: This "parse as JSON else default" behaviour is used enough times that it should probably be extracted to a helper function.

File: .docker/scripts/init-ext.sh [hunk 4]
```diff
@@ -27,11 +27,22 @@ install_requirements () {
 
 . ${APP_DIR}/scripts/activate
 
-install_requirements . dev-requirements requirements-dev
+install_requirements . extensions
```
Comment (ThrawnCA) [inline .docker/scripts/init-ext.sh:30]: This doesn't match usual naming conventions at all. Why was it changed?

**Review by ThrawnCA — CHANGES_REQUESTED** [review]

**Comment by duttonw** [comment]
OpenData Team have accepted the business risk. Yes there is things in there we would like done differently as it is causing tech debt, overall it is still in better shape than last year. (except for the need for a test plugin shim that is interacted by test fixtures reading 'tags')

**Pull request merged**